lib/src/widgets/board_editor.dart

```diff
@@ -0,0 +1,274 @@
+import 'package:chessground/src/board_geometry.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+
+import 'piece.dart';
+import 'positioned_square.dart';
+import '../models.dart';
+import '../fen.dart';
+import '../board_settings.dart';
+
+class BoardEditor extends StatefulWidget with BoardGeometry {
+  const BoardEditor({
+    super.key,
+    required this.size,
+    required this.orientation,
+    required this.initialFen,
+    this.settings = const BoardSettings(),
+  });
+
+  /// Visual size of the board.
+  @override
+  final double size;
+
+  /// Settings that control the theme, behavior and purpose of the board.
+  final BoardSettings settings;
+
+  /// Side by which the board is oriented.
+  @override
+  final Side orientation;
+
+  /// FEN string describing the initial position of the board.
+  final String initialFen;
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _BoardEditorState createState() => _BoardEditorState();
+}
+
+class _BoardEditorState extends State<BoardEditor> {
+  Pieces pieces = {};
+
+  /// Avatar for the piece that is currently being dragged.
+  DragAvatar? _dragAvatar;
+
+  /// Once a piece is dragged, holds the square id of the piece.
+  SquareId? _draggedPieceSquareId;
+
+  /// Current pointer down event.
+  ///
+  /// This field is reset to null when the pointer is released (up or cancel).
+  ///
+  /// This is used to track board gestures, the pointer that started the drag,
+  /// and to prevent other pointers from starting a drag while a piece is being
+  /// dragged.
+  ///
+  /// Other simultaneous pointer events are ignored and will cancel the current
+  /// gesture.
+  PointerDownEvent? _currentPointerDownEvent;
+
+  /// Current render box during drag.
+  // ignore: use_late_for_private_fields_and_variables
+  RenderBox? _renderBox;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = widget.settings.colorScheme;
+
+    final background = widget.settings.enableCoordinates
+        ? widget.orientation == Side.white
+            ? colorScheme.whiteCoordBackground
+            : colorScheme.blackCoordBackground
+        : colorScheme.background;
+
+    final List<Widget> pieceWidgets = [
+      for (final entry in pieces.entries)
+        if (entry.key != _draggedPieceSquareId)
+          PositionedSquare(
+            key: ValueKey('${entry.key}-${entry.value.kind.name}'),
+            size: widget.squareSize,
+            orientation: widget.orientation,
+            squareId: entry.key,
+            child: PieceWidget(
+              piece: entry.value,
+              size: widget.squareSize,
+              pieceAssets: widget.settings.pieceAssets,
+              blindfoldMode: widget.settings.blindfoldMode,
+            ),
+          ),
+    ];
+
+    return Listener(
```

Comment:
For the editor it may be better to use a `GestureDetector` instead of the `Listener`. It will be easier to deal with.

An example of the `Board` implemented with gesture detector can be found in the commit history.

Reply:
Thanks for the hint, will take a look